pandas/plotting/_matplotlib/style.py

```diff
@@ -1,81 +1,163 @@
-                # GH17525 use common._random_state to avoid resetting the seed
-                rs = com.random_state(column)
-                return rs.rand(3).tolist()
-
-            colors = [random_color(num) for num in range(num_colors)]
-        else:
-            raise ValueError("color_type must be either 'default' or 'random'")
+        return _get_colors_from_color_type(color_type, num_colors=num_colors)
+
 
-    if isinstance(colors, str) and _is_single_color(colors):
-        # GH #36972
-        colors = [colors]
+def _cycle_colors(colors: List[Color], num_colors: int) -> List[Color]:
+    """Append more colors by cycling if there is not enough color.
 
-    # Append more colors by cycling if there is not enough color.
-    # Extra colors will be ignored by matplotlib if there are more colors
-    # than needed and nothing needs to be done here.
+    Extra colors will be ignored by matplotlib if there are more colors
+    than needed and nothing needs to be done here.
+    """
     if len(colors) < num_colors:
-        try:
-            multiple = num_colors // len(colors) - 1
-        except ZeroDivisionError:
-            raise ValueError("Invalid color argument: ''")
+        multiple = num_colors // len(colors) - 1
         mod = num_colors % len(colors)
-
         colors += multiple * colors
         colors += colors[:mod]
 
     return colors
 
 
+def _get_colors_from_colormap(
+    colormap: Union[str, "Colormap"],
+    num_colors: int,
+) -> Collection[Color]:
+    """Get colors from colormap."""
+    colormap = _get_cmap_instance(colormap)
+    return [colormap(num) for num in np.linspace(0, 1, num=num_colors)]
+
+
+def _get_cmap_instance(colormap: Union[str, "Colormap"]) -> "Colormap":
+    """Get instance of matplotlib colormap."""
+    if isinstance(colormap, str):
+        cmap = colormap
+        colormap = cm.get_cmap(colormap)
+        if colormap is None:
+            raise ValueError(f"Colormap {cmap} is not recognized")
+    return colormap
+
+
+def _get_colors_from_color(
+    color: Union[Color, Dict[str, Color], Collection[Color]],
+) -> Union[Dict[str, Color], Collection[Color]]:
+    """Get colors from user input color."""
+    if isinstance(color, Iterable) and len(color) == 0:
+        raise ValueError("Invalid color argument: {color}")
+
+    if isinstance(color, dict):
+        return color
+
+    if isinstance(color, str):
+        if _is_single_color(color):
+            # GH #36972
+            return [color]
+        else:
+            return list(color)
+
+    # ignoring mypy error here
+    # error: Argument 1 to "list" has incompatible type
+    # "Union[Sequence[float], Collection[Union[str, Sequence[float]]]]";
+    # expected "Iterable[Union[str, Sequence[float]]]"  [arg-type]
+    # A this point color may be sequence of floats or series of colors,
+    # all convertible to list
+    return list(color)  # type: ignore [arg-type]
```

Comment:
the ci failure could be fixed with 
```suggestion
    return list(color)  # type: ignore[arg-type]
```
but please avoid using ignore instead.